firmware/controllers/sensors/flex_sensor.cpp

```diff
@@ -0,0 +1,37 @@
+#include "engine.h"
+#include "digital_input_exti.h"
+#include "flex_sensor.h"
+
+EXTERN_ENGINE;
+
+static void flexExtiCallback(void* arg) {
+	auto inst = reinterpret_cast<FlexFuelSensor*>(arg);
+	inst->onEdge(getTimeNowNt());
+}
+
+void FlexFuelSensor::init(brain_pin_e pin) {
+#if EFI_PROD_CODE
+	// todo: refactor https://github.com/rusefi/rusefi/issues/2123
+	efiExtiEnablePin("flex", pin, 
+		PAL_EVENT_MODE_FALLING_EDGE,
+		flexExtiCallback, reinterpret_cast<void*>(this));
+#endif // EFI_PROD_CODE
+}
+
+FlexFuelSensor::FlexFuelSensor()
+	// this long timeout is ok, ethanol content isn't going to change quickly
+	: StoredValueSensor(SensorType::FuelEthanolPercent, MS2NT(500))
+{
+}
+
+void FlexFuelSensor::onEdge(efitick_t nowNt) {
+	float frequency = 1 / m_edgeTimer.getElapsedSecondsAndReset(nowNt);
+
+	// TODO: are these limits reasonable?
+	if (frequency > 45 && frequency < 155) {
```

Comment:
Limits might be a bit broad, most people seem to settle at 49hz and 151hz for floor and ceil.

And signals above 175hz indicate a fault condition ( fuel containmination to trigger failsafe ), we can probably add it as a TODO

Reply:
Good to know. Doesn't really hurt to be a little generous here, as 0.3% is a little tight on the high end (though E100 is also rare).

Reply:
Yup, we should be fine with these values 👍 LGTM!